### PR TITLE
WT-11333 Update test/format to generate random compaction "free_space_target" values for foreground compaction

### PR DESCRIPTION
**Pull request by Sean04**

### CODE REVIEW

File: test/format/format_config_def.c [hunk 1]
```diff
@@ -175,6 +175,9 @@ CONFIG configuration_list[] = {{"assert.read_timestamp", "assert read_timestamp"
   {"format.major_timeout", "long-running operations timeout (minutes)", C_IGNORE, 0, 0, WT_THOUSAND,
     V_GLOBAL_FORMAT_MAJOR_TIMEOUT},
 
+  {"free_space_target", "configure free space target for compaction", 0x0, 1, 100, UINT_MAX,
```
Comment (pmacko86) [inline test/format/format_config_def.c:178]: You should explicitly mention the unit in the description (i.e., "MB").

**Review by pmacko86 — APPROVED** [review]
LGTM, after fixing one small issue (and the Evergreen test failures).

File: test/format/format_config.c [hunk 2]
```diff
@@ -36,6 +36,7 @@ static void config_cache(void);
 static void config_checkpoint(void);
 static void config_checksum(TABLE *);
 static void config_compression(TABLE *, const char *);
+static void config_compact_free_space_target(void);
```
Comment (etienneptl) [inline test/format/format_config.c:39]: Not sure if this one is auto-generated but it does not seem to be A->Z

**Review by etienneptl — APPROVED** [review]

**Comment by Sean04** [comment]
Test format is failing due to the config check missing for the new API. I will wait for WT-11332 before merging this.

**Review by etienneptl — APPROVED** [review]
Nice one!

**Comment by Sean04** [comment]
evergreen merge

**Pull request merged**